server/src/main/java/com/broadcom/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -213,11 +224,13 @@ private void interruptAnalysis(String uri) {
   }
 
   @Override
+  @CheckServerShutdownState
   public void didSave(DidSaveTextDocumentParams params) {
     LOG.info("Document saved...");
   }
 
   @Override
+  @CheckServerShutdownState
```

Comment:
Not an LSP request, no need to check it

Reply:
I spent major time here. Using interceptor actually leads to some proxy classes and it seems without this annotation here, this method is marked final which leads to error while spying. I get the below error

> java.lang.VerifyError: class com.broadcom.lsp.cobol.service.CobolTextDocumentService$$EnhancerByGuice$$c86ee31e$MockitoMock$565924183 overrides final method com.broadcom.lsp.cobol.service.CobolTextDocumentService$$EnhancerByGuice$$c86ee31e.observerCallback(Ljava/lang/Object;)

This confuses me if guice interceptors are the right approach.  This is a question to all reviewers, if you feel the same , I will re-work this with some other approach.
For now, I have kept this annotation as is.